src/main/scala/abalone/Pos.scala

```diff
@@ -194,7 +194,7 @@ case class Pos private (index: Int) extends AnyVal {
   def neighbours: List[Option[Pos]] = List(left, upLeft, upRight, right, downRight, downLeft)
   def neighboursAsDirs: Directions = List(_.left, _.upLeft, _.upRight, _.right, _.downRight, _.downLeft)
 
-  def dir(dest: Pos): String =
+  def directionString(dest: Pos): String =
```

Comment:
Finally getting round to reviewing this...
So I would prefer you to create a type for `directionString` rather than have it return `String`. Your code is 'Stringly Typed' (google it). It would be better and safer if Strings weren't used for these directions. You can create a type, probably in this file which then explicitly lists out the directions. it can have a single parameter which is a String, but when handling one of these directions you couldn't accidentally pass in "not-a-direction-but-a-string" as a parameter. I also note that there seems to be two sets of directions, one that uses all 6 directions of movement on a hexagonal board, and one that it used for side moves, which only has diagonals (so 4). Using a better typing system could ensure these couldnt be muddled up

Reply:
thank you for the explanation